ospf6d/ospf6_top.c

```diff
@@ -950,67 +950,108 @@ DEFUN (show_ipv6_ospf6,
 	return CMD_SUCCESS;
 }
 
-DEFUN (show_ipv6_ospf6_route,
-       show_ipv6_ospf6_route_cmd,
-       "show ipv6 ospf6 route [<intra-area|inter-area|external-1|external-2|X:X::X:X|X:X::X:X/M|detail|summary>]",
-       SHOW_STR
-       IP6_STR
-       OSPF6_STR
-       ROUTE_STR
-       "Display Intra-Area routes\n"
-       "Display Inter-Area routes\n"
-       "Display Type-1 External routes\n"
-       "Display Type-2 External routes\n"
-       "Specify IPv6 address\n"
-       "Specify IPv6 prefix\n"
-       "Detailed information\n"
-       "Summary of route table\n")
+DEFUN(show_ipv6_ospf6_route,
+      show_ipv6_ospf6_route_cmd,
+      "show ipv6 ospf6 route [<intra-area|inter-area|external-1|external-2|X:X::X:X|X:X::X:X/M|detail|summary>] [json]",
+      SHOW_STR
+      IP6_STR
+      OSPF6_STR
+      ROUTE_STR
+      "Display Intra-Area routes\n"
+      "Display Inter-Area routes\n"
+      "Display Type-1 External routes\n"
+      "Display Type-2 External routes\n"
+      "Specify IPv6 address\n"
+      "Specify IPv6 prefix\n"
+      "Detailed information\n"
+      "Summary of route table\n"
+      JSON_STR)
 {
+	bool uj = use_json(argc, argv);
+	json_object *json = NULL;
+
 	OSPF6_CMD_CHECK_RUNNING();
 
-	ospf6_route_table_show(vty, 4, argc, argv, ospf6->route_table);
+	if (uj) {
+		json = json_object_new_object();
+		ospf6_route_table_show(vty, 4, argc, argv, ospf6->route_table,
+				       json, uj);
+		vty_out(vty, "%s\n",
+			json_object_to_json_string_ext(
+				json, JSON_C_TO_STRING_PRETTY));
+		json_object_free(json);
+	} else
+		ospf6_route_table_show(vty, 4, argc, argv, ospf6->route_table,
+				       json, uj);
 	return CMD_SUCCESS;
 }
 
 DEFUN (show_ipv6_ospf6_route_match,
        show_ipv6_ospf6_route_match_cmd,
-       "show ipv6 ospf6 route X:X::X:X/M <match|longer>",
+       "show ipv6 ospf6 route X:X::X:X/M <match|longer> [json]",
        SHOW_STR
        IP6_STR
        OSPF6_STR
        ROUTE_STR
        "Specify IPv6 prefix\n"
        "Display routes which match the specified route\n"
-       "Display routes longer than the specified route\n")
+       "Display routes longer than the specified route\n"
+       JSON_STR)
 {
+	bool uj = use_json(argc, argv);
+	json_object *json = NULL;
+
 	OSPF6_CMD_CHECK_RUNNING();
 
-	ospf6_route_table_show(vty, 4, argc, argv, ospf6->route_table);
+	if (uj) {
```

Comment:
since this block of code is used at multiple places, can you move this into a function and call from everywhere else or move the logic inside ospf6_route_table_show function. argc and arg are being passed already to the function.